modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2AuthorizationCodeGrantJWTTokenTestCase.java

```diff
@@ -332,6 +334,45 @@ public void testValidateAdditionalUserClaims() {
         });
     }
 
+    @Test(groups = "wso2.is", description = "Call introspect endpoint", dependsOnMethods = "testExtractJWTAccessTokenClaims")
+    public void testIntrospectRefreshToken() throws Exception {
+
+        List<NameValuePair> urlParameters = new ArrayList<>();
+        urlParameters.add(new BasicNameValuePair("token", refreshToken));
+
+        List<Header> headers = new ArrayList<>();
+        headers.add(new BasicHeader(AUTHORIZATION_HEADER, OAuth2Constant.BASIC_HEADER + " " +
+                getBase64EncodedString(isServer.getContextTenant().getTenantAdmin().getUserName(),
+                        isServer.getContextTenant().getTenantAdmin().getPassword())));
+        headers.add(new BasicHeader("Content-Type", "application/x-www-form-urlencoded"));
+        headers.add(new BasicHeader("User-Agent", OAuth2Constant.USER_AGENT));
+
+        HttpResponse response = sendPostRequest(client, headers, urlParameters,
+                getTenantQualifiedURL(INTRO_SPEC_ENDPOINT, tenantInfo.getDomain()));
+
+        assertNotNull(response, "Failed to receive a response for introspection request.");
+        assertEquals(response.getStatusLine().getStatusCode(), HttpStatus.SC_OK,
+                response.getStatusLine().getReasonPhrase());
+
+        String responseString = EntityUtils.toString(response.getEntity(), "UTF-8");
+        EntityUtils.consume(response.getEntity());
+        JSONObject jsonResponse = new JSONObject(responseString);
+
+        assertTrue(jsonResponse.has("nbf"), "Not Before value not found in the introspection response");
+        assertTrue(jsonResponse.has("exp"), "Expiry timestamp not found in the introspection response");
+        long exp = jsonResponse.getLong("exp");
+        assertTrue(jsonResponse.has("iat"), "Issued at timestamp not found in the introspection response");
+        long iat = jsonResponse.getLong("iat");
+
+        assertEquals((exp - iat), applicationConfig.getRefreshTokenExpiryTime(),
+                "Invalid expiry time for the refresh token.");
+
+        assertTrue(jsonResponse.has("scope"), "Scopes not found in the introspection response");
```

Comment:
shouldn't we validate scopes ?

Reply:
Fixed 1fd25850cba1120b2d58f0ac007de71ed127e4db